packages/babel-compiler/babel-compiler.js

```diff
@@ -51,7 +53,9 @@ BCp.processFilesForTarget = function (inputFiles) {
 // Returns an object suitable for passing to inputFile.addJavaScript, or
 // null to indicate there was an error, and nothing should be added.
 BCp.processOneFileForTarget = function (inputFile, source) {
+  var self = this; // capture context
```

Comment:
Why do we need to introduce a new variable called "self" instead of using "this"? I understand it's quite common in Python, but it's not something you typically see in the JavaScript ecosystem.

Reply:
I had to save this in a variable `self` to preserve the context. Inside profile('Babel.compile', function () {, this refers to a different context, while we still want to access the one above.

Reply:
an option is to use bind or arrow functions like

``` js
function f1() {
    try { 
        var result = profile('Babel.compile', function cb() {
            console.log(this) // will use f1's context
        }.bind(this))
    }
    // ...
}

// or

function f1() {
    try { 
        var result = profile('Babel.compile', () => { 
            console.log(this) // lexically binds to f1's context
        })
    }
    // ...
}

Reply:
Since I refactored with `Babel.compile` and `SWC.compile` profile wrappers, there’s no need for self anymore. I switched to an arrow function, which takes care of the this binding. Unless there are additional comments, I’ll consider this resolved.